### PR TITLE
[IMP] [12.0] [agreement_legal] Amend demo data with new fields #306

### LINKED ISSUE #306: [IMP][12.0] agreement_legal: Amend demo data with new fields
The "demo" data loaded by the "agreement" module, should be amended with values for the new fields in agreement_legal. Especially "description" and "type" which are mandatory fields.

### PR DESCRIPTION
**Pull request by ygol**
fix #306

### CODE REVIEW

File: agreement_legal/__manifest__.py [hunk 1]
```diff
@@ -18,6 +18,7 @@
         "data/module_category.xml",
         "data/agreement_stage.xml",
         "data/agreement_type.xml",
+        "data/demo.xml",
```
Comment (max3903) [inline agreement_legal/__manifest__.py]: @ygol This should be added in the `demo` keyword.

File: agreement_legal/data/demo.xml [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+                eval="[('module', '=', 'agreement'), ('name', '=', 'market1')]"
+            />
+        </function>
+
+        <value eval="{'noupdate': False}" />
+    </function>
+
+    <record id="agreement.market1" model="agreement">
+        <field name="description">Hardware IT (C2C-IT0042)</field>
+        <field name="agreement_type_id">1</field>
```
Comment (max3903) [inline agreement_legal/data/demo.xml]:
```suggestion
        <field name="agreement_type_id" ref="agreement_legal.agreement_type_agreement"/>
```

File: agreement_legal/data/demo.xml [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+                eval="[('module', '=', 'agreement'), ('name', '=', 'market2')]"
+            />
+        </function>
+
+        <value eval="{'noupdate': False}" />
+    </function>
+
+    <record id="agreement.market2" model="agreement">
+        <field name="description">Fiber access office Lausanne (C2C-IT0043)</field>
+        <field name="agreement_type_id">2</field>
```
Comment (max3903) [inline agreement_legal/data/demo.xml]: @ygol Please use the ref.

File: agreement_legal/data/demo.xml [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+                eval="[('module', '=', 'agreement'), ('name', '=', 'market3')]"
+            />
+        </function>
+
+        <value eval="{'noupdate': False}" />
+    </function>
+
+    <record id="agreement.market3" model="agreement">
+        <field name="description">Vétérinaire (AGR-VETO001)</field>
+        <field name="agreement_type_id">3</field>
```
Comment (max3903) [inline agreement_legal/data/demo.xml]: @ygol Same thing.

File: agreement_legal/data/demo.xml [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+
+    <function name="write" model="ir.model.data">
+        <function name="search" model="ir.model.data">
+            <value
+                eval="[('module', '=', 'agreement'), ('name', '=', 'market1')]"
+            />
+        </function>
+
+        <value eval="{'noupdate': False}" />
+    </function>
```
Comment (max3903) [inline agreement_legal/data/demo.xml]: @ygol I am not sure to get the purpose of this block. Can you explain please?

File: agreement_legal/__manifest__.py [hunk 6]
```diff
@@ -18,6 +18,7 @@
         "data/module_category.xml",
         "data/agreement_stage.xml",
         "data/agreement_type.xml",
+        "demo/demo.xml",
```
Comment (max3903) [inline agreement_legal/__manifest__.py]: @ygol I meant something like this:
https://github.com/OCA/maintainer-tools/blob/master/template/module/__manifest__.py#L38

File: agreement_legal/data/demo.xml (hunk 5 above)
Comment (ygol) [reply agreement_legal/data/demo.xml]: It is not needed indeed. tnx

File: agreement_legal/__manifest__.py [hunk 7]
```diff
@@ -11,7 +11,7 @@
     "website": "https://github.com/OCA/contract",
     "category": "Partner",
     "license": "AGPL-3",
-    "version": "12.0.1.0.0",
+    "version": "12.0.1.0.1",
```
Comment (max3903) [inline agreement_legal/__manifest__.py]:
```suggestion
    "version": "12.0.1.1.0",
```

The changes require a module upgrade to be effective. A server restart would have no effect:
https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#version-numbers

**Review by max3903 — APPROVED** [review]

**Review by osi-scampbell — APPROVED** [review]

**Pull request merged**